File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel

    def subscribed
        stream_from 'chat_channel'
        ChatBroadcastJob.perform_later
    end

    def unsubscribed
        stop_all_streams
    end

    # def speak(data)
    #     # ActionCable.server.broadcast 'chat_channel', message: data['message']
    #     # current_user.messages.create!(body: data['message'])
    # end
end

File: app/javascript/channels/chat_channel.js
import consumer from "./consumer"

document.addEventListener('turbolinks:load', function() {
    console.log('i upload')
    const chat = document.querySelectorAll('#chat')
    if (chat.length > 0) {
        consumer.subscriptions.create("ChatChannel", {

            connected() {
                console.log('connect')
            },

            disconnected() {

            },

            received(data) {
                let chatMessages = chat[0].querySelector('.chat-messages')
                chatMessages.insertAdjacentHTML('beforeEnd', data)
            },

            // speak: function(message) {
            //     console.log('i speaked')
            //     return this.perform('speak', message);
            // }
        });
    }
})

